dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreCollectionCommandBuilder.cs

```diff
@@ -410,7 +406,23 @@ private static (List<string> Columns, List<string> ParameterNames, List<object?>
 
     private static string GetParameterName(string propertyName, int index)
     {
-        return $"@{propertyName}{index}";
+        StringBuilder builder = new();
```

Comment:
Was there a bug here that this fixes? Or is this about slightly better perf?

Reply:
The input value (`propertyName`) can be column name with a character that is not valid for parameter names:

https://github.com/microsoft/semantic-kernel/blob/6bcce23f85eac9d204a1503c230168cd0a3814c2/dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreCollectionCommandBuilder.cs#L395-L399

The tests I've provided cover it (when I revert this change, they fail)